### PR TITLE
fix mixed sum/mean losses. Added compensation for smaller batches. Ad…

### PR DESCRIPTION
**Pull request by randommm**
fix mixed sum/mean losses. Added compensation for smaller batches. Added new network architecture. Rename to package ir to lowercase

### CODE REVIEW

File: survival4d/nn/torch/__init__.py [hunk 1]
```diff
@@ -41,7 +41,7 @@ def train_nn(xtr, ytr, batch_size, n_epochs, model_name, lr_exp, alpha, weight_d
     model = model_factory(model_name, **model_kwargs)
 
     dataset = TensorDataset(torch.from_numpy(X_tr).cuda(), torch.from_numpy(E_tr).cuda(), torch.from_numpy(TM_tr).cuda())
-    dataloader = DataLoader(dataset, batch_size=batch_size, shuffle=False)
+    dataloader = DataLoader(dataset, batch_size=batch_size, shuffle=False, drop_last=True)
```
Comment (lisurui6) [inline survival4d/nn/torch/__init__.py:44]: I am not sure that we should drop last

File: survival4d/nn/torch/__init__.py [hunk 2]
```diff
@@ -52,18 +52,26 @@ def train_nn(xtr, ytr, batch_size, n_epochs, model_name, lr_exp, alpha, weight_d
...
             neg_log = negative_log_likelihood(e, risk_pred)
             loss = mse_loss * alpha + (1 - alpha) * neg_log
+
             optimizer.zero_grad()
-            loss.backward()
+            loss_corrected = loss
+            actual_batch_size = x.shape[1]
+            if actual_batch_size != batch_size:
+                loss_corrected = loss_corrected * actual_batch_size / batch_size
+            loss_corrected.backward()
```
Comment (lisurui6) [inline survival4d/nn/torch/__init__.py:68]: loss are already mean values, so doing loss backward and loss corrected backward should be the same.
